### PR TITLE
feat: set mgr to interface

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolve: https://github.com/envoyproxy/gateway/issues/182

Since this is a backlog issue, I will work on this after releasing v0.2.0.

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

**Comment by LukeShu** [comment]
Looks OK to me, but I know that @danehans was hesitant to create an `interface` until we had a second provider to have an idea of what things might need to be different between providers.

**Comment by Xunzhuo** [comment]
Thanks for this comment @LukeShu, yes, so this can wait until we got another provider, and we can do better abstraction.

File: internal/infrastructure/manager.go [hunk 1]
```diff
@@ -10,26 +11,30 @@ import (
 	"github.com/envoyproxy/gateway/internal/envoygateway"
 	"github.com/envoyproxy/gateway/internal/envoygateway/config"
 	"github.com/envoyproxy/gateway/internal/infrastructure/kubernetes"
+	"github.com/envoyproxy/gateway/internal/ir"
 )
 
+var _ Manager = (*kubernetes.Infra)(nil)
```
Comment (arkodg) [inline internal/infrastructure/manager.go:17]: why is this needed ?

**Comment by arkodg** [comment]
should be fine to get this in, and iterate, thanks for adding this @Xunzhuo

File: internal/infrastructure/manager.go (hunk 1 above)
Comment (Xunzhuo) [reply internal/infrastructure/manager.go:17]: Just make sure kubernetes.Infra have implemented the interface.:)

**Review by arkodg — APPROVED** [review]
